packages/core/database/lib/utils/boolean.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const parseBoolean = value => {
```

Comment:
we export this function from @strapi/utils I believe. if you are moving it out I guess better just import it from the strapi utils ?

Reply:
Yes we can, however it is not exactly the same.
In `@strapi/utils` it is:
```js
const parseType = ({ type, value }) => {
  switch (type) {
    case 'boolean': {
      if (typeof value === 'boolean') return value;

      if (['true', 't', '1', 1].includes(value)) {
        return true;
      }

      if (['false', 'f', '0', 0].includes(value)) {
        return false;
      }

      throw new Error('Invalid boolean input. Expected "t","1","true","false","0","f"'); // <- here is the difference
    }
    case 'integer':
    case 'biginteger':
    case 'float':
    case 'decimal': {
      return _.toNumber(value);
    }
    case 'time': {
      return parseTime(value);
    }
    case 'date': {
      return parseDate(value);
    }
    case 'timestamp':
    case 'datetime': {
      return parseDateTimeOrTimestamp(value);
    }
    default:
      return value;
  }
};
```

I can use it but we need to decide if we throw an error or if we cast to boolean anyway. What do you think?

Reply:
We don't want to make a breaking change so it depends what we currently do. I think we won't be able to throw for now. However we can split part of the logic to have the parseBoolean function and only throw in parseType